- kdevelop-5.6.2.orig/cmake/modules/FindClang.cmake   2021-02-01 16:30:08.000000000 +0100
+++ kdevelop-5.6.2/cmake/modules/FindClang.cmake        2024-03-26 14:00:28.516566220 +0100
@@ -102,12 +102,17 @@
   # dito for git
   string(REPLACE "git" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
 
+  # get major version to find clang version 17
+  string(REPLACE "." ";" VERSION_LIST ${CLANG_VERSION})
+  list(GET VERSION_LIST 0 CLANG_VERSION_MAJOR)
+
   find_path(CLANG_BUILTIN_DIR
             # cpuid.h because it is defined in ClangSupport constructor as valid clang builtin dir indicator
             NAMES "cpuid.h"
             PATHS "${CLANG_LIBRARY_DIRS}"
                   "${CLANG_INCLUDE_DIRS}"
             PATH_SUFFIXES "clang/${CLANG_VERSION}/include"
+                          "clang/${CLANG_VERSION_MAJOR}/include"
                           "../../../clang/${CLANG_VERSION}/include"
                           "clang/${CLANG_VERSION_CLEAN}/include"
                           "../../../clang/${CLANG_VERSION_CLEAN}/include"
